src/sentry/release_health/metrics.py

```diff
@@ -265,99 +277,98 @@ def _get_release_adoption_impl(
-                Condition(Column("timestamp"), Op.LT, now),
             ]
 
             if environments is not None:
                 where_common.append(
                     Condition(
-                        Column(resolve_tag_key(USE_CASE_ID, org_id, "environment")),
-                        Op.IN,
-                        resolve_many_weak(USE_CASE_ID, org_id, environments),
+                        lhs=Column("tags[environment]"),
+                        op=Op.IN,
+                        rhs=environments,
                     )
                 )
 
             if not total:
-                where_common.append(filter_releases_by_project_release(org_id, project_releases))
+                where_common.append(filter_releases_by_project_release(project_releases))
 
             return where_common
 
-        def _get_common_groupby(total: bool) -> List[SelectableExpression]:
+        def _get_common_groupby(total: bool) -> List[MetricGroupByField]:
             if total:
-                return [Column("project_id")]
+                return [MetricGroupByField(field="project_id")]
             else:
                 return [
-                    Column("project_id"),
-                    Column(resolve_tag_key(USE_CASE_ID, org_id, "release")),
+                    MetricGroupByField(field="project_id"),
+                    MetricGroupByField(field="release"),
                 ]
 
-        def _convert_results(data: Any, total: bool) -> Dict[Any, int]:
-            if total:
-                return {x["project_id"]: x["value"] for x in data}
-            else:
-                release_tag = resolve_tag_key(USE_CASE_ID, org_id, "release")
-                return {(x["project_id"], x[release_tag]): x["value"] for x in data}
+        def _convert_results(groups: Any, total: bool) -> Dict[Any, int]:
+            """
+            Converts the result groups into an array of values:
 
-        def _count_sessions(total: bool, referrer: str) -> Dict[Any, int]:
-            query = Query(
-                match=Entity(EntityKey.MetricsCounters.value),
-                select=[Function("sum", [Column("value")], "value")],
-                where=_get_common_where(total)
-                + [
-                    Condition(
-                        Column("metric_id"),
-                        Op.EQ,
-                        resolve(USE_CASE_ID, org_id, SessionMRI.SESSION.value),
-                    ),
-                    Condition(
-                        Column(resolve_tag_key(USE_CASE_ID, org_id, "session.status")),
-                        Op.EQ,
-                        resolve_weak(USE_CASE_ID, org_id, "init"),
-                    ),
-                ],
+            from [{ "by": {"project_id": 123, "release": "r1"}, "totals": {"init": 23.3}},...]
+            to:
+             { 123: 23.3, ...} // for totals
+             { (123, "r1"): 23.3, ...} // for details
+
+            """
+            ret_val = {}
```

Comment:
nit: Can we name it like `result`?

Reply:
`ret_val` = return value... I am using it everywhere I touch the code for many years so it would be a rather large refactoring :-).

Reply:
yeah ret_val - is good name. Even better than result.